modules/openapi-generator/src/main/java/org/openapitools/codegen/OpenAPINormalizer.java

```diff
@@ -1268,6 +1268,7 @@ private Schema processNormalize31Spec(Schema schema, Set<Schema> visitedSchemas)
                 as.setMaxItems(schema.getMaxItems());
                 as.setExtensions(schema.getExtensions());
                 as.setXml(schema.getXml());
+                as.setNullable(schema.getNullable());
```

Comment:
Further up, we fix up the `nullable` flag when the types contain the new 3.1 `null` type: https://github.com/OpenAPITools/openapi-generator/blob/29c3b020e620330669bac9950863b09bc2e61764/modules/openapi-generator/src/main/java/org/openapitools/codegen/OpenAPINormalizer.java#L1249-L1252

but then, if there's a single type left AND it is an array schema, we forget to copy this property to the cloned `ArraySchema`. This line makes sure that the nullable flag is transferred onto the schema clone.

Reply:
This issue is likely to happen again in the future when/if a new property is added that is in 3.1+ but not 3.0.x - we could possibly introduce a `clone` or `transfer` function somewhere which would make this a bit more obvious, but it's not a guarantee either.